### PR TITLE
Add macOS support for CI using GitHub Actions

### PR DESCRIPTION
**Pull request by sergeyklay**
- Move ZendEngine3 kernel to the separate project
- Add macOS support on GitHub Actions

### CODE REVIEW

File: .github/workflows/build-linux.yml [hunk 1]
```diff
@@ -8,16 +8,21 @@ on:
...
+    name: "${{ matrix.os }}: PHP v${{ matrix.php }}"
+    runs-on: ${{ matrix.os }}
 
     strategy:
       fail-fast: false
 
       matrix:
+        os:
+          - ubuntu-18.04
+          - macOS-latest
```
Comment (ruudboon) [inline .github/workflows/build-linux.yml]: OS can be lowercase I think.

File: unit-tests/Extension/MCallDynamicTest.php [hunk 2]
```diff
@@ -21,7 +21,8 @@ public function callDynamic()
     {
         $a = new McallDynamic();
         $this->assertSame(1, $a->method1());
-        $this->assertSame(2, $a->testMagicCall1());
+        // FIXME: Does not work on macOs
```
Comment (ruudboon) [inline unit-tests/Extension/MCallDynamicTest.php:24]: Maybe create an issue as well and link it in the comments?

File: .github/workflows/build-linux.yml (hunk 1 above)
Comment (ruudboon) [reply .github/workflows/build-linux.yml]: See release notes: https://github.com/shivammathur/setup-php/releases

Comment (sergeyklay) [reply .github/workflows/build-linux.yml]: @ruudboon Somehow it works. Did you see? :)

File: unit-tests/Extension/MCallDynamicTest.php (hunk 2 above)
Comment (sergeyklay) [reply unit-tests/Extension/MCallDynamicTest.php:24]: Work not done yet, I plan to continue investigate on this. No separate issue required

File: .github/workflows/build-linux.yml (hunk 1 above)
Comment (ruudboon) [reply .github/workflows/build-linux.yml]: Haha yes. For now

**Review by ruudboon — APPROVED** [review]

**Pull request closed without merging**